students/yixingxu/lesson05/mailroom3.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+
+# added exception handling in this version
+
+import sys # import sys, so i can use sys.exit(0) later
+import io
+donation_history = [
+                    {'name':'Anna','donations':[100,200,300]},
+                    {'name':'Bob','donations':[1000,2000]},
+                    {'name':'Chuck','donations':[10000]},
+                    {'name':'David','donations':[1]},
+                    {'name':'Ethan','donations':[10,20]}
+                 ]
```

Comment:
another data structure to make things easier:
```
donors = {"Anna": [100, 200, 300], "Bob": [2000]}
```
donor's first name is a natural key in this case so it makes sense to make it the dict key since it will make it easier for us to find the donor.